builder/utils/deployer.py

```diff
@@ -426,7 +447,8 @@ def deploy(self, s3_uri):
             self.cleanup_on_failure(s3_uri)
             ex_msg = "Unexpected error"
         else:
-            self.cleanup_post_deployment()
+            if not endpoint_only:
+                self.cleanup_post_deployment()
```

Comment:
This is because if we are bringing up something that is `endpoint_only=True`, then we never build the image, so there is not image to delete (it does not crash if we do run `cleanup_post_deployment()` it just says `No image found`)